tests/filtering/test_spectrum_processor.py

```diff
@@ -120,10 +119,9 @@ def test_filter_spectrums_report(spectrums):
     assert report.counter_changed_metadata == {'make_charge_int': 2, 'interpret_pepmass': 3, 'derive_ionmode': 3}
     report_df = report.to_dataframe()
     assert np.all(report_df.loc[["require_minimum_number_of_peaks", "interpret_pepmass",
-                                 "add_losses", "correct_charge"]].values == np.array(
+                                 "correct_charge"]].values == np.array(
         [[1, 0, 0],
-         [0, 3, 0],
-         [0, 0, 2],
+         [0, 3, 3],
```

Comment:
@niekdejonge I didn't get the number is this array or why this changed more than one line after I removed `add_losses`. Can you briefly check if that's correct?

Reply:
Hmm I tried it in the main branch locally. And for me the test passes when kept as [0,3,0]. I only changed this test, not the rest of the code. So that is surprising to me as well.

Reply:
@florian-huber I don't know what is causing this. I have made the same changes to the test in #683 Do you have an idea what could be the issue?

Reply:
This is weird. In two ways. 
A) Interpret_pepmass should only change the metadata, and
B) Why does it change 3 spectra anyway? After the `require_minimum...` filter in the test data there should only be 2 spectra left, right?

```
s1 = SpectrumBuilder(). \
    with_metadata({"charge": "+1", "pepmass": 100}). \
    with_mz([10, 20, 30]).with_intensities([0.1, 0.4, 10]).build()
s2 = SpectrumBuilder().with_metadata({"charge": "-1",
                                      "pepmass": 102}). \
    with_mz([10, 20, 30]).with_intensities([0.1, 0.2, 1]).build()
s3 = SpectrumBuilder().with_metadata({"charge": -1,
                                      "pepmass": 104}). \
    with_mz([10, ]).with_intensities([0.1, ]).build()
spectrums = [s1, s2, s3]

processor = SpectrumProcessor(filters=["make_charge_int",
                                       "interpret_pepmass",
                                       "derive_ionmode",
                                       "correct_charge",
                                       ])
processor.parse_and_add_filter(filter_description=("require_minimum_number_of_peaks", {"n_required": 2}))
spectrums, report = processor.process_spectrums(spectrums)
```

Reply:
Discussion on slack solved this:
- require filter runs after the others (my bet)
- and, losses are still tested in the report --> will be removed.

Reply:
@niekdejonge fyi: I just adapted this.